clang/lib/Parse/ParseDecl.cpp

```diff
@@ -6750,6 +6750,21 @@ void Parser::ParseDeclaratorInternal(Declarator &D,
     // Is a reference
     DeclSpec DS(AttrFactory);
 
+    // Complain about parsing ambiguity of parsing reference types
+    // inside expressions with reflection in C++2x
+    if (D.getContext() == DeclaratorContext::ReflectOperator &&
+        !Tok.isOneOf(tok::r_paren, tok::semi)) {
+      // parser already consumed '^' token before setting this context
+      assert((Kind == tok::ampamp || Kind == tok::amp) &&
```

Comment:
Unfortunately, there are a lot more cases here. I'm not sure this approach will work (although it might - not really sure one way or the other).

```cpp
r == ^int& ? 1 : 0
condition ? ^int& : ^int;
^int&(*)(bool)
^int& != r
^int& == r
constexpr auto r {^int&};
```

In addition to those, in the presence of some user-defined
```cpp
consteval result_type operator???(std::meta::info R, MyStupidClass);
```

You could have `^int& + o`, along with: `-`, `*`, `/`, `%`, `]`, `|`, `||` `<=`, `>=`, `>`.

Those are all the cases I can think of off the top of my head, but I may have forgotten some.

Reply:
thank you! gonna think more how to address it